packages/serialize-request/lib/index.js

```diff
@@ -1,5 +1,23 @@
 /**
- * @typedef {import('express').Request | import('http').IncomingMessage & {route: object, params: object}} Request
+ * @typedef {import('express').Request} ExpressRequest
+ */
+
+/**
+ * @typedef {import('http').IncomingMessage} HttpIncomingMessage
+ */
+
+/**
+ * @typedef {object} BasicRequest
+ * @property {Object<string, string>} headers
+ *     The request HTTP headers.
+ * @property {string} method
+ *     The request HTTP method.
+ * @property {string} url
+ *     The request URL.
```

Comment:
**suggestion**: you're checking these properties exist already, so i think it would make sense to make them optional in this type? would help with compatibility with Apollo's request type, where they're also optional

Reply:
Yeah that makes sense, technically any can be missing and we'll still try and serialize _something_. Done 👍

Reply:
i see the types are optional in the copypasta in the description above but they're not in the PR diff, are you waiting for me to test it before committing?

Reply:
Oh they are sorry, I force pushed which confused things. In JSDoc square braces around a property name is the equivalent of `?` in TypeScript:

```js
/**
 * @typedef {object} Whatever
 * @property {string} [optionalProp]
 * @property {string} nonOptionalProp
 */
```

The new `BasicRequest` type is:

```js
/**
 * @typedef {object} BasicRequest
 * @property {Object<string, string>} [headers]
 *     The request HTTP headers.
 * @property {string} [method]
 *     The request HTTP method.
 * @property {string} [url]
 *     The request URL.
 */
```

Reply:
ohhhhhh